pxr/usdValidation/usdUtilsValidators/testenv/testUsdUtilsValidators.cpp

```diff
@@ -113,11 +126,102 @@ TestPackageEncapsulationValidator()
     TF_AXIOM(errors.empty());
 }
 
+static void
+TestMissingReferenceValidator()
+{
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidationValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdUtilsValidatorNameTokens->missingReferenceValidator);
+
+    TF_AXIOM(validator);
+
+    const UsdStageRefPtr& stage = UsdStage::CreateInMemory();
+
+    const UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/Xform"));
+    const SdfReference badLayerReference("doesNotExist.usd");
+    xform.GetPrim().GetReferences().AddReference(badLayerReference);
+
+    // Validate an error occurs on a missing layer reference
+    {
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify both the layer errors are present
+        const std::string expectedErrorMessage = "Found unresolvable external "
+                                                 "dependency 'doesNotExist.usd'.";
+        const TfToken expectedIdentifier =
+            TfToken(
+            "usdUtilsValidators:MissingReferenceValidator.UnresolvableDependency");
+        TF_AXIOM(errors.size() == 1);
+
+        ValidateError(errors[0], expectedErrorMessage,
+                expectedIdentifier);
+    }
+
+    xform.GetPrim().GetReferences().RemoveReference(badLayerReference);
+    const SdfReference badAssetReference("doesNotExist.jpg");
+    xform.GetPrim().GetReferences().AddReference(badAssetReference);
```

Comment:
Ahh don't think its legal to have a non-usd as a reference like this.

An asset would be something like this: https://github.com/PixarAnimationStudios/OpenUSD/blob/9b0c13b2efa6233c8a4a4af411833628c5435bde/extras/usd/tutorials/simpleShading/simpleShading.usda#L48

So what you can do is:
```python
>>> from pixar import Sdf, Usd, UsdShade
>>> stage = Usd.Stage.CreateInMemory()
>>> shader = UsdShade.Shader.Define(stage, "/Shader")
>>> assetInput = shader.CreateInput("notFoundAsset", Sdf.ValueTypeNames.Asset)
>>> assetInput.Set("NotFoundTexture.png")
True
>>> stage.ExportToString()
'#usda 1.0\n(\n    doc = """Generated from Composed Stage of root layer \n"""\n)\n\ndef Shader "Shader"\n{\n    asset inputs:notFoundAsset = @NotFoundTexture.png@\n}\n\n'
>>>
```

Reply:
Thanks! This has been updated